### DISCUSSION TITLE
I don't understand how to log many to many relationship properly

### DISCUSSION THREAD

**Discussion by Oui-Dev** [spatie/laravel-activitylog#1269] [Q&A]
Hi everyone, i have tried many things to be able to log many to many relationships but i keep struggling to do this as clean as possible.
The best way i found is this [#487 (comment)](https://github.com/spatie/laravel-activitylog/issues/487#issuecomment-1766234260) but as you can see in my answer that's not perfect, in some cases that doesn't work.

If someone has a better way, I'm more than interested 🙏

**Comment by AmirKhan47** [comment]
any solutions?

**Reply by Oui-Dev** [reply]
I think this library is dead...
There hasn't been any major change for months, even years, even 4.8 just adds support for Laravel 11 and updating dependencies... Nobody responds to help requests or issues, and I don't think a 5.0 is in development.
My patient is at her limits and I have to finish this task soon, so I'm thinking of removing this library and implementing my own activity tracking system...

If you find a better solution, I'm interested 🙂.
